Source/Common/IGListBatchUpdateData.mm

```diff
@@ -29,16 +29,45 @@ static void convertMoveToDeleteAndInsert(NSMutableSet<IGListMoveIndex *> *moves,
 
 @implementation IGListBatchUpdateData
 
+/**
+ Finds duplicate index paths and increments increments the item index so eacy operation is unique. Duplicate delete
+ operations cannot be performed within the same batch update block. However deletes have corresponding data-source
+ changes so the operation still has to occur, otherwise UICollectionView will throw an inconsistency exception. See
+ https://github.com/Instagram/IGListKit/issues/651
+ */
+static NSIndexPath *staggeredIndexPath(NSCountedSet<NSIndexPath *> *dupes, NSIndexPath *path) {
+    const NSInteger count = [dupes countForObject:path];
+    if (count == 0) {
+        [dupes addObject:path];
+        return path;
+    }
+
+    NSIndexPath *nextPath = [NSIndexPath indexPathForItem:path.item + count inSection:path.section];
+    NSIndexPath *staggeredPath = staggeredIndexPath(dupes, nextPath);
+    [dupes addObject:staggeredPath];
+    return staggeredPath;
```

Comment:
👍 will add some docs to this.

I think the struct will get really messy w/ the recursion tho